src/qiskit_qec/utils/decoding_graph_attributes.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Graph used as the basis of decoders.
+"""
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
+
+from qiskit_qec.exceptions import QiskitQECError
+
+
+class DecodingGraphNode:
+    """
+    Class to describe DecodingGraph nodes.
+
+    Attributes:
+     - is_boundary (bool): whether or not the node is a boundary node.
+     - time (int): what syndrome node the node corrsponds to. Doesn't
+        need to be set if it's a boundary node.
+     - qubits (List[int]): List of indices which are stabilized by
+        this ancilla.
+     - index (int): Unique index in measurement round.
+     - properties (Dict[str, Any]): Decoder/code specific attributes.
+        Are not considered when comparing nodes.
+    """
+
+    def __init__(self, qubits: List[int], index: int, is_boundary=False, time=None) -> None:
```

Comment:
Could we make `qubits` optional? Sometimes we might make decoders for qudit codes, for example, so this won't always be relevant.